backend/src/rtp.rs

```diff
@@ -1730,25 +1789,25 @@ impl Endpoint {
     // Returns parsed and decrypted RTCP, and also processes transport-cc feedback based on
     // previously sent packets (if they had transport-cc seqnums).
     // Also decrypts the packet in place.
-    // TODO: Use Result instead of Option.
     pub fn receive_rtcp(
         &mut self,
         encrypted: &mut [u8],
         now: Instant,
-    ) -> Option<ProcessedControlPacket> {
+    ) -> Result<ProcessedControlPacket, EndpointError> {
         let incoming = ControlPacket::parse_and_decrypt_in_place(
             encrypted,
             &self.decrypt.rtcp.key,
             &self.decrypt.rtcp.salt,
-        )?;
+        )
+        .ok_or(EndpointError::RtcpSenderSsrcNotSet)?;
```

Comment:
I think `parse_and_decrypt_in_place` should return EndpointError? They are all in the same file and hence should be able to share Error definitions. You could add all the different kinds of RTCP errors that the function might return. That might increase the scope of the PR a lot. Hmm, is it worth it?

Maybe just make the error more generic, like `RtcpError`. Because errors are more than just "Sender SSRC was not set", right?

Reply:
I think making the error type generic is a nice idea I initially thought about it though, but didn't proceeded with that due to the over complexity it might present in maintaining it in the long run hence I opted for writing errors enum for each struct possible and keeping minimum only.

Also maybe we can add a another module for specifically for handling scenarios only( all errors seperated in that module only). 

Again we need some more brainstorming on what can work out best and avoiding the burden, but speaking of generic errors I can happily work on this if you can present with some rough idea of what errors and the possible variants it should have for the scenarios !   

just to explain what I understood is written below:

```rust

#[derive(...)]
pub enum RTXErrors {...
}
#[derive(...)]
pub enum SsrcErrors{...
}
#[derive(...)]
pub enum PacketErrors {...
}
#[derive(...)]
pub enum EndpointError<T> {...
```

to visually explain it, here is a small diagram I have prepared
![image](https://github.com/signalapp/Signal-Calling-Service/assets/45896902/1d073441-9927-4b8a-8790-a6006ee7efc9)

Reply:
I am not sure that many error types is necessary. For example, I only see one RTXError, so I think it can just stay as an EndpointError. For SsrcError, that looks okay.